SnM/SnM_FX.cpp

```diff
@@ -101,14 +102,37 @@ bool SetOrToggleFXBypassSelTracks(const char* _undoMsg, int _mode, int _fxCmdId,
 							}
 						}
 						break;
-					case 2: // toggle
-						if (fxId < fxcnt)
+					case 2: // toggle selected fx
+					{
+						// Get handle to FX window for this track
+						HWND fxChain = CF_GetTrackFXChain(tr);
+						if(fxChain)
 						{
-							if (_undoMsg && !updated)
-								Undo_BeginBlock2(NULL);
-							TrackFX_SetEnabled(tr, fxId, !TrackFX_GetEnabled(tr, fxId));
-							updated = true;
+							int index=-1;
+							// enumerate selected FX
+							while( (index = CF_EnumSelectedFX(fxChain, index)) >= 0)
+							{
+								if (_undoMsg && !updated)
+									Undo_BeginBlock2(NULL);
+								// toggle enabled state
+								TrackFX_SetEnabled(tr, index, !TrackFX_GetEnabled(tr, index));
+								updated = true;
+							}
 						}
+						else if(fxId < fxcnt)
+						{
+							int index=-1;
+							// enumerate selected FX
+							while( (index = CF_EnumSelectedFX(fxChain, index)) >= 0)
```

Comment:
`fxChain` is NULL at this point, so the action fails to find the last active FX when the chain is closed. The last active FX index is stored in `fxId`.

Reply:
> `fxChain` is NULL at this point, so the action fails to find the last active FX when the chain is closed. The last active FX index is stored in `fxId`.
> 
Strange, I have the same code in both blocks, but I remember testing this with FX window opened and closed.  Anyhow, sorry, I pushed an update that reverts to the previous behavior if the FX window is not opened.

> Updating the meaning of "selected FX" from "the last FX shown in the chain" to "all selected FXs in the chain" is interesting, but I think it should be done across all SWS actions dealing with "selected FX". [#1432 (comment)](https://github.com/reaper-oss/sws/pull/1432#issuecomment-739555373)

IIUC the original request was to make the action behave the same as the menu command (Ctrl-B).  I searched for other actions that affected selected FX, are these what you were wanting to include?:

SWS/S&M: Show FX chain for selected tracks (selected FX)
SWS/S&M: Float selected FX for selected tracks
SWS/S&M: Unfloat selected FX for selected tracks
SWS/S&M: Toggle float selected FX for selected tracks
SWS/S&M: Move selected FX up in chain for selected tracks
SWS/S&M: Move selected FX down in chain for selected tracks
SWS/S&M: Remove selected FX for selected tracks
SWS/S&M: Toggle selected FX online/offline for selected tracks
SWS/S&M: Set selected FX online for selected tracks
SWS/S&M: Set selected FX offline for selected tracks
SWS/S&M: Bypass selected FX for selected tracks
SWS/S&M: Unbypass selected FX for selected tracks
SWS/S&M: Trigger preset for selected FX of selected track (MIDI/OSC only)
SWS/S&M: Trigger next preset for selected FX of selected tracks
SWS/S&M: Trigger previous preset for selected FX of selected tracks
SWS/S&M: Reassign MIDI learned channels of selected FX for selected tracks (prompt)